File: Arjuna/versions/PR3.3.3_98_01_10/ObjectStore/src/StoreList.cc
/*
 * Copyright (C) 1993
 *
 * Department of Computing Science,
 * The University,
 * Newcastle upon Tyne,
 * UK.
 *
 * $Id: StoreList.cc,v 
 */

#ifndef STORELIST_H_
#  include "StoreList.h"
#endif

StoreList::StoreList ()
                      : useCount(0),
			instance(0),
			next(0)
{
}

StoreList::~StoreList ()
{
    if (instance)
	delete instance;
}

File: JavaArjuna/v0.1.0/classes/com/arjuna/JavaArjuna/ObjectStore/StoreList.java
/*
 * Copyright (C) 1996, 1997, 1998,
 *
 * Department of Computing Science,
 * University of Newcastle upon Tyne,
 * Newcastle upon Tyne,
 * UK.
 *
 * $Id: StoreList.java,v 1.3 1998/07/06 13:26:48 nmcl Exp $
 */

package com.arjuna.JavaArjuna.ObjectStore;

/*
 * Default visibility.
 */

class StoreList
{

public StoreList ()
    {
	useCount = 0;
	instance = null;
	next = null;
    }

public long useCount;
public ObjectStoreImple instance;
public StoreList next;
    
}
